lib/Sema/CSSimplify.cpp

```diff
@@ -6034,36 +6052,137 @@ ConstraintSystem::simplifyCheckedCastConstraint(
       break;
   } while (true);
 
+  auto attemptFixForCheckedConvertibleTypes = [&]() -> ConstraintFix * {
+    // Both types have to be fixed.
+    if (fromType->hasTypeVariable() || toType->hasTypeVariable())
+      return nullptr;
+
+    SmallVector<LocatorPathElt, 4> path;
+    auto anchor = locator.getLocatorParts(path);
+
+    // Don't record any fix when recursive simplifying for Dictionary, Set or
+    // Array element types.
+    if (!path.empty() &&
+        path.back().is<LocatorPathElt::KnownStdlibCollectionElement>())
```

Comment:
I wonder if instead of adding this overly specific locator element it would be enough to pass `TMF_ApplyingFix` as a flag to `simplifyCheckedCastConstraint` when it's called for collection element types?

Reply:
Fixed 👍